content/en/blog/_posts/2022-11-11-retroactive-default-storage-class.md

```diff
@@ -0,0 +1,131 @@
+---
+layout: blog
+title: "Retroactive default StorageClass"
+date: 2022-11-11
+slug: retroactive-default-storage-class
+---
+
+**Author:** Roman Bednář (Red Hat)
+
+In Kubernetes 1.25 a new Alpha feature was introduced to improve the way default StorageClass is assigned to a PersistentVolumeClaim (PVC). It is no longer required to create a default StorageClass first and PVC second in order to assign the class and any PVCs without StorageClass can be updated later. The feature will be Beta in Kubernetes 1.26.
+
+See [Official Kubernetes documentation](https://kubernetes.io/docs/concepts/storage/persistent-volumes/#retroactive-default-storageclass-assignment) for more details.
+
+## Why did StorageClass assignment need improvements
+
+When talking about PVCs and default StorageClasses you might think "It already works, when I create PVC it gets the default class just fine!" and that is correct because there already is a feature that causes admission controller to assign a class to **new** PVCs which is documented [here](https://kubernetes.io/docs/reference/access-authn-authz/admission-controllers/#defaultstorageclass).
+
+But what if there was no default StorageClass at the time of PVC creation? Users end up with a PVC that will never get a class and so no storage will be provisioned, the PVC is somewhat "stuck" at this point. From high level point of view there were two main use cases which might result in "stuck" PVCs and cause problems later down the road. Let's take a closer look at each of them.
+
+### Changing default StorageClass did not have a good solution
+
+   There are two options admins had when they wanted to change default StorageClass:
+   1. Creating the new StorageClass as default before removing the old one which results in having two defaults for a short period of time. If a user creates a PersistentVolumeClaim at this point with storageClassName set to `nil` (which means default StorageClass) Kubernetes 1.26+ will choose the newest default StorageClass and assign it to this PVC.
+
+
+   2. Remove the old default first and create new default StorageClass which results in having no default for a short period of time. If user creates a PersistentVolumeClaim with storageClassName set to `nil` (which means default StorageClass) the PVC will be in `Pending` state forever, then user has to fix this by deleting the PVC a creating it again once default StorageClass is available.
```

Comment:
These lines should not have leading whitespace. Also, please wrap the source code - it makes the work of localization teams much easier.

Reply:
Done. You mean wrap into `<code>` tag instead of ``` ` ``` right?